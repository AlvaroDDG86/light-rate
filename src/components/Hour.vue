<template>
    <div class="inline-block hour m-2 animate__animated animate__bounceIn">
        <div
            class="bg-indigo-50 px-8 py-8 bg-white w-96 max-w-xs overflow-hidden rounded-lg shadow-md bg-white hover:shadow-xl transition-shadow duration-300 ease-in-out"
            :class="{ 'border-pink-400 border-2': active }"
        >
            <h3 class="py-2 text-4xl font-bold font-mono p-4 rounded-md" :class="getBGColor()">{{ value }} <span class="text-2xl">â‚¬/MWh</span></h3>
            <div class="text-center mt-2 leading-none flex justify-between w-full">
                <span class="mr-3 inline-flex items-center leading-none text-2xl py-1 font-bold">
                  {{ percent }}
                </span>
                <span class=" inline-flex items-center leading-none text-5xl font-bold">
                  {{ hour }}
                </span>
            </div>
        </div>
    </div>
</template>
<script>
export default {
    props: {
        data: {
            type: Object,
            required: true,
            default () {
                return {
                    value: '',
                    percentage: '',
                    datetime: ''
                }
            }
        },
        active: {
            type: Boolean,
            default: false
        }
    },
    computed: {
        hour() {
            return !this.data ? '' : new Date(this.data.datetime).toLocaleTimeString().slice(0, -3)
        },
        percent() {
            return !this.data ? '' :  `${parseFloat(this.data.percentage).toFixed(2)} %`
        },
        value() {
            return !this.data ? '' : parseFloat(this.data.value).toFixed(2)
        }
    },
    methods: {
        getBGColor() {
            if (this.data) {
                return {
                    'bg-red-400': (this.data.value > 220),
                    'bg-yellow-200': (this.data.value < 220 && this.data.value > 150),
                    'bg-green-400': (this.data.value < 150)
                }
            }
        }
    }
}
</script>
<style lang="scss" scoped>
</style>