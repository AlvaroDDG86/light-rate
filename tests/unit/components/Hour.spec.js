import { shallowMount } from '@vue/test-utils'
import Hour from '@/components/Hour'

const factory = (options) => {
    return shallowMount(Hour, {
        ...options
    })
}

describe('Hour', () => {
    test('should match with snapshot', () => {
        const wrapper = factory({
            props: {
                data: {
                    datetime: "2022-02-12T09:00:00.000+01:00",
                    percentage: 0.4427229653234936,
                    value: 211.17
                }
            },
            active: true
        })
        expect(wrapper.html()).toMatchSnapshot()
    });
});