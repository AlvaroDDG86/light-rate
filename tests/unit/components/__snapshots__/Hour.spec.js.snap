// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hour should match with snapshot 1`] = `
<div class="inline-block hour m-2 animate__animated animate__bounceIn">
  <div class="bg-indigo-50 px-8 py-8 bg-white w-96 max-w-xs overflow-hidden rounded-lg shadow-md bg-white hover:shadow-xl transition-shadow duration-300 ease-in-out">
    <h3 class="py-2 text-4xl font-bold font-mono p-4 rounded-md"> <span class="text-2xl">â‚¬/MWh</span></h3>
    <div class="text-center mt-2 leading-none flex justify-between w-full"><span class="mr-3 inline-flex items-center leading-none text-2xl py-1 font-bold">

            </span> <span class=" inline-flex items-center leading-none text-5xl font-bold">

            </span></div>
  </div>
</div>
`;
